src/Microsoft.Identity.Client/TokenCache.cs

```diff
@@ -260,7 +260,9 @@ internal RefreshTokenCacheItem FindRefreshToken(AuthenticationRequestParameters
         {
             lock (LockObject)
             {
-                TokenCacheKey key = new TokenCacheKey(null, null, requestParam.ClientId, requestParam.User?.HomeObjectId);
+                RefreshTokenCacheKey key = new RefreshTokenCacheKey(
+                    new Uri(requestParam.Authority.CanonicalAuthority).Host, requestParam.ClientId,
```

Comment:
can't the authority class expose a method i.e. getHost instead of the new Uri used in multiple places?